src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/InitializeHandler.cs

```diff
@@ -18,7 +18,7 @@ internal class InitializeHandler : IRequestHandler<InitializeParams, InitializeR
             {
                 CompletionProvider = new CompletionOptions()
                 {
-                    AllCommitCharacters = new[] { " ", ".", ";", ">", "=" }, // This is necessary to workaround a bug where the commit character in CompletionItem is not respected.
+                    AllCommitCharacters = new[] { " ", ".", ";", ">", "=", ":" }, // This is necessary to workaround a bug where the commit character in CompletionItem is not respected.
```

Comment:
@ajaybhargavb why does this impact our Razor language server completion endpoint?

Reply:
That is a great question. When I suggested this, I didn't realize this was in our endpoint. Now that I think about it, it shouldn't have affected this. Very interesting.

Reply:
If this is actually impacting our Razor language endpoint this is a pretty awful multi-language server bug and will bite us harddd once we start including more features.

Reply:
Could it be that our HtmlCSharp server is calling the Razor server manually and that is why this is respected? @TanayParikh, to test this, you can just return null in `CompletionHandler` and see if you still get the same behavior for directive attributes. Ideally, that should still work because `CompletionHandler` only takes care of HTML and CSharp completion.

Reply:
Returned `null` from https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs#L63

No longer getting completions.

Reply:
>No longer getting completions.

That's a problem. We should still be getting Tag helper completions from `RazorCompletionEndpoint`. This could explain why adding a `:` here impacted that.

Reply:
I'm surprised. Our projection provider shouldn't be invoking anything in our Razor language server: https://github.com/dotnet/aspnetcore-tooling/blob/b0e378c1626ea7c43d29cc4cbe32ccf3f0d50bcd/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPProjectionProvider.cs#L70

Reply:
Note, we have two `onkeypress`s.
1. HTML `onkeypress`
2. Razor Directive Attribute `@onkeypress`

For either to complete with a `:` commit char, we need to add the commit char here. Adding it to the `RazorLanguageServer`: https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs#L155-L160 does not work.

Reply:
Created an issue to track this: https://github.com/dotnet/aspnetcore/issues/21346